util/edns0.go

```diff
@@ -0,0 +1,130 @@
+package util
+
+import (
+	"slices"
+
+	"github.com/miekg/dns"
+)
+
+// EDNS0Option is an interface for all EDNS0 options as type constraint for generics.
+type EDNS0Option interface {
+	*dns.EDNS0_SUBNET | *dns.EDNS0_EDE | *dns.EDNS0_LOCAL | *dns.EDNS0_NSID | *dns.EDNS0_COOKIE | *dns.EDNS0_UL
+	Option() uint16
+}
+
+// RemoveEdns0Record removes the OPT record from the Extra section of the given message.
+func RemoveEdns0Record(msg *dns.Msg) bool {
+	if msg == nil || msg.IsEdns0() == nil || len(msg.Extra) == 0 {
+		return false
+	}
+
+	for i, rr := range msg.Extra {
+		if rr.Header().Rrtype == dns.TypeOPT {
+			msg.Extra = slices.Delete(msg.Extra, i, i+1)
+
+			return true
+		}
+	}
+
+	return false
+}
+
+// GetEdns0Record returns the OPT record from the Extra section of the given message.
+func GetEdns0Record(msg *dns.Msg) *dns.OPT {
+	if msg == nil {
+		return nil
+	}
+
+	if res := msg.IsEdns0(); res != nil {
+		return res
+	}
+
+	res := new(dns.OPT)
+	res.Hdr.Name = "."
+	res.Hdr.Rrtype = dns.TypeOPT
+	msg.Extra = append(msg.Extra, res)
+
+	return res
+}
```

Comment:
I think a `Get` function shouldn't create the option if missing. Can we move that part to `SetEdns0Option` and here always return `msg.IsEdns0()`?  
Maybe we can do without the function if it's just that though.

Reply:
Maybe renaming it would be an option.

Since there should only be one EDNS0 OPT record I thought it would be handy to have one function which always returns this record.😅. 
Always checking for it's existence and creating it if it's missing would possibly create code redundancy....

Reply:
Yeah but if we have it in `SetEdns0Option`, we still wouldn't need that pattern IIUC?

Reply:
You're right.

Since we have utility functions to directly manipulate the options this function is technically obsolete.

I also considered moving it to another package since util is a bit generic but wasn't sure about the name.  
Since `dns` would often interfer with other libraries... maybe `dnsutil`? 🤔

Reply:
I think leaving it here for now is fine. We have a bunch of other stuff that should go in such a package so I think it'd be best to do that separately.